src/plone/rest/traverse.py

```diff
@@ -14,35 +12,22 @@
 @adapter(ISiteRoot, IAPIRequest)
 class RESTTraverse(DefaultPublishTraverse):
     def publishTraverse(self, request, name):
-        try:
-            obj = super(RESTTraverse, self).publishTraverse(request, name)
-            if not IContentish.providedBy(obj) and not IService.providedBy(obj):
-                if isinstance(obj, VirtualHostMonster):
-                    return obj
-                else:
-                    raise KeyError
-        except KeyError:
-            # No object, maybe a named rest service
-            service = queryMultiAdapter(
-                (self.context, request), name=request._rest_service_id + name
-            )
-            if service is None:
-                # No service, fallback to regular view
-                view = queryMultiAdapter((self.context, request), name=name)
-                if view is not None:
-                    return view
-                raise
+
+        service = queryMultiAdapter(
+            (self.context, request), name=request._rest_service_id + name
+        )
+        if service:
             return service
 
-        if name.startswith(request._rest_service_id):
-            return obj
-
-        # Do not handle view namespace
-        if "@@" in request["PATH_INFO"] or "++view++" in request["PATH_INFO"]:
-            return obj
+        obj = super(RESTTraverse, self).publishTraverse(request, name)
 
         # Wrap object to ensure we handle further traversal
-        return RESTWrapper(obj, request)
+        if IContentish.providedBy(obj) and not (
```

Comment:
I am not sure why we check for IContentish here, rather than ITraversable? I tried that and it seems to solve https://github.com/plone/plone.restapi/issues/680 and https://github.com/plone/plone.app.theming/issues/165 whilst still passing at least `plone.rest` tests.

Reply:
Interesting. I'd guess that IContentish is always ITraversable. IContentish excludes the SiteRoot, not sure if that is traversable. Things are complex and different people worked on different parts that depend on plone.rest, so I'd like them to weight in as well. cc @buchi @jaroel @sneridagh

Reply:
I actively remove `IContentish` from `Plone Site` in https://github.com/plone/Products.CMFPlone/pull/2612/files#diff-860754a57fed2533e4aae6aab34a27caR238-R242 . I did not check how that will affect `plone.rest` stuff, but so far it seemed to work.
My gut reaction is that `ITraversable` is correct here.

Reply:
We're adding IContentish back to PloneSite.
I don't think this changes my position on ITraversable, but it probably will affect other stuff.

Is this whole branch/approach still relevant actually?